collections/map_values.ts

```diff
@@ -26,15 +26,51 @@
  * );
  * ```
  */
-export function mapValues<T, O>(
-  record: Readonly<Record<string, T>>,
+export function mapValues<T, O, K extends string>(
+  record: Readonly<Record<K, T>>,
   transformer: (value: T) => O,
-): Record<string, O> {
-  const ret: Record<string, O> = {};
+): Record<K, O>;
```

Comment:
Should this overload be deprecated?

Reply:
That makes this change a breaking change (`Partial<Record>` is not assignable to `Record`). Also that makes this util less convenient if the input is actually `Record` in my view (`record[key]` is `O` for `Record<K, O>` but, it's `O | undefined` for `Partial<Record<K, O>>`)